From cb88b51916626888708dc93514069c8fd023cbca Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Fri, 29 May 2020 17:56:10 +0200
Subject: [PATCH] PONRTSYS-6699: Check if rules list is empty when deleting
 vlan storage node

This is to check if the rule list is really empty when the rule_cnt
reached 0, but we checked the wrong list before and this warning got
always triggered.

Fixes: c8b9330b85a2 ("PONRTSYS-6699:")
---
 drivers/net/datapath/pon_qos/pon_qos_tc_vlan_storage.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_storage.c b/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_storage.c
index 5777112eab9f..e634ab8c6414 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_storage.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_storage.c
@@ -58,8 +58,8 @@ void pon_qos_tc_vlan_storage_del(struct net_device *dev,
 	if (node->rule_cnt > 0)
 		return;
 
-	if (!list_empty(&node->list))
-		netdev_warn(dev, "%s: list not empty!\n", __func__);
+	if (!list_empty(&node->rules))
+		netdev_warn(dev, "%s: rules list not empty!\n", __func__);
 
 	netdev_dbg(dev, "VLAN storage deleted: %p\n", node);
 	list_del(&node->list);
